File: daily_python/1288_Remove_Covered_Intervals.py
import os
import sys
from typing import List

class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x : (x[0], -x[1]))
        curr = 0
        d = 0
        for s, e in intervals:
            if e <= curr:
                d += 1
            curr = max(curr, e)
        return len(intervals) - d

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            intervals = [list(map(int, interval.split(','))) for interval in lines[0].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append(intervals)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for intervals in dataset:
        results.append(solution.removeCoveredIntervals(intervals))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/1288_Remove_Covered_Intervals.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        auto cmp = [](const vector<int>& a, const vector<int>& b){
            if (a[0] == b[0]) return a[1] > b[1];
            return a[0] < b[0];
        };
        sort(intervals.begin(), intervals.end(), cmp);
        int ans = 0;
        int currMax = 0;
        for (auto& interval : intervals){
            if (interval[1] <= currMax) ans++;
            currMax = max(currMax, interval[1]);
        }
        return intervals.size() - ans;
    }
};

int main(){
    vector<vector<int>> intervals = {{1,4},{3,6},{2,8}};
    // vector<vector<int>> intervals = {{1,4},{2,3}};
    Solution S;
    cout << S.removeCoveredIntervals(intervals) << endl;
    return 0;
}